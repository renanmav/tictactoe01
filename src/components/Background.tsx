import React from 'react';
import { Dimensions, ImageBackground, StatusBar } from 'react-native';
import { t } from 'react-native-tailwindcss';

const s = {
  background: [t.itemsCenter, t.justifyCenter],
};

const { width, height } = Dimensions.get('window');

const Background: React.FC<{ testID: string }> = ({ children, testID }) => {
  return (
    <ImageBackground
      source={require('../../assets/background.jpg')}
      style={[s.background, { width, height }]}
      testID={testID}>
      <StatusBar barStyle="light-content" />
      {children}
    </ImageBackground>
  );
};

export default Background;
