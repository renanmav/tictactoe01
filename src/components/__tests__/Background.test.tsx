import React from 'react';
import { render } from '@testing-library/react-native';

import Background from '../Background';
import { View } from 'react-native';

it('should render children', async () => {
  const { getByTestId, baseElement } = render(
    <Background>
      <View testID="content" />
    </Background>
  );

  expect(getByTestId('content')).toBeDefined();
  expect(baseElement).toMatchSnapshot();
});
