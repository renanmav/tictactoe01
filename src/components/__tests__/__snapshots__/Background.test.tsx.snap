// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render children 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "justifyContent": "center",
            },
          ],
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        testID="content"
      />
    </View>
  </View>
</View>
`;
